$NetBSD: patch-ab,v 1.11 2013/10/27 20:41:49 joerg Exp $

Caution: the changes for using a different cpp for munging the
makefiles (search for MAKEFILECPP) are *not* reflected in the pkgsrc
patch for configure.in; don't accidentally lose them by regenerating
with autoconf without thinking.

--- configure.orig	2000-06-12 20:05:56.000000000 +0000
+++ configure
@@ -20,6 +20,8 @@ ac_help="$ac_help
 ac_help="$ac_help
   --with-x-toolkit=KIT    use an X toolkit (KIT = yes/lucid/athena/motif/no)"
 ac_help="$ac_help
+  --with-ipv6             support IPv6 stack"
+ac_help="$ac_help
   --with-x                use the X Window System"
 
 # Initialize some variables set by options.
@@ -584,6 +586,12 @@ Currently, \`yes', \`athena' and \`lucid
 
 fi
 
+# Check whether --with-ipv6 or --without-ipv6 was given.
+if test "${with_ipv6+set}" = set; then
+  withval="$with_ipv6"
+  :
+fi
+
 
 #### Make srcdir absolute, if it isn't already.  It's important to
 #### avoid running the path through pwd unnecessary, since pwd can
@@ -712,15 +720,20 @@ case "${canonical}" in
     opsys=netbsd
     case "${canonical}" in
       alpha*-*-netbsd*)	machine=alpha ;;
+      hppa*-*-netbsd*)	machine=hp800 ;;
       i[3456]86-*-netbsd*) machine=intel386 ;;
       m68k-*-netbsd*)
 			# This is somewhat bogus.
 			machine=hp9000s300 ;;
-      powerpc-apple-netbsd*) machine=macppc ;;
       mips-*-netbsd*)	machine=pmax ;;
+      mipsel-*-netbsd*)	machine=pmax ;;
+      mipsev-*-netbsd*)	machine=mips ;;		# HF: what else?
       ns32k-*-netbsd*)	machine=ns32000 ;;
-      sparc-*-netbsd*)	machine=sparc ;;
+      powerpc-*-netbsd*) machine=macppc ;;
+      sparc*-*-netbsd*)	machine=sparc ;;
       vax-*-netbsd*)	machine=vax ;;
+      arm*-*-netbsd*)	machine=arm ;;
+      x86_64-*-netbsd*)	machine=amd64 ;;
     esac
   ;;
 
@@ -1519,6 +1532,7 @@ case "${canonical}" in
       *-sysv4.2uw* )	  	opsys=unixware; NON_GNU_CPP=/lib/cpp ;;
       *-386bsd* )	        opsys=386bsd ;;
       *-freebsd* )	        opsys=freebsd ;;
+      *-dragonfly* )		opsys=dragonfly ;;
       *-nextstep* )             opsys=nextstep ;;
       ## Otherwise, we'll fall through to the generic opsys code at the bottom.
     esac
@@ -1981,6 +1995,14 @@ else
 fi
 echo "$ac_t""$CPP" 1>&6
 
+echo $ac_n "checking how to preprocess makefiles""... $ac_c" 1>&6
+echo "configure:1999: checking how to preprocess makefiles" >&5
+# If we were doing this for real we could test for various alternatives
+# (note the now-dead version 1.4 of patch-ab for some things to check)
+# but for pkgsrc purposes the following is currently sufficient.
+
+echo "$ac_t""$MAKEFILECPP" 1>&6
+
 # Find a good install program.  We prefer a C program (faster),
 # so one script is as good as another.  But avoid the broken or
 # incompatible versions:
@@ -3823,6 +3845,53 @@ else
   echo "$ac_t""no" 1>&6
 fi
 
+  if test x$HAVE_X11XTR6 = xyes; then
+    LIBS="$LIBS -lXpm"
+    echo $ac_n "checking for XawPixmapsInitialize in -lXaw""... $ac_c" 1>&6
+echo "configure:3842: checking for XawPixmapsInitialize in -lXaw" >&5
+ac_lib_var=`echo Xaw'_'XawPixmapsInitialize | sed 'y%./+-%__p_%'`
+if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
+  echo $ac_n "(cached) $ac_c" 1>&6
+else
+  ac_save_LIBS="$LIBS"
+LIBS="-lXaw  $LIBS"
+cat > conftest.$ac_ext <<EOF
+#line 3850 "configure"
+#include "confdefs.h"
+/* Override any gcc2 internal prototype to avoid an error.  */
+/* We use char because int might match the return type of a gcc2
+    builtin and then its argument prototype would still apply.  */
+char XawPixmapsInitialize();
+
+int main() {
+XawPixmapsInitialize()
+; return 0; }
+EOF
+if { (eval echo configure:3861: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
+  rm -rf conftest*
+  eval "ac_cv_lib_$ac_lib_var=yes"
+else
+  echo "configure: failed program was:" >&5
+  cat conftest.$ac_ext >&5
+  rm -rf conftest*
+  eval "ac_cv_lib_$ac_lib_var=no"
+fi
+rm -f conftest*
+LIBS="$ac_save_LIBS"
+
+fi
+if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
+  echo "$ac_t""yes" 1>&6
+   LIBXPM="-lXpm" 
+else
+  echo "$ac_t""no" 1>&6
+ LIBXPM=""
+fi
+
+  else
+    LIBXPM=""
+  fi
+  
   LIBS="$OLDLIBS"
 fi
 
@@ -4906,6 +4975,64 @@ EOF
 
 fi
 
+if test "x$with_ipv6" = xyes; then
+  for ac_func in getaddrinfo
+do
+echo $ac_n "checking for $ac_func""... $ac_c" 1>&6
+echo "configure:4973: checking for $ac_func" >&5
+if eval "test \"`echo '$''{'ac_cv_func_$ac_func'+set}'`\" = set"; then
+  echo $ac_n "(cached) $ac_c" 1>&6
+else
+  cat > conftest.$ac_ext <<EOF
+#line 4978 "configure"
+#include "confdefs.h"
+/* System header to define __stub macros and hopefully few prototypes,
+    which can conflict with char $ac_func(); below.  */
+#include <assert.h>
+/* Override any gcc2 internal prototype to avoid an error.  */
+/* We use char because int might match the return type of a gcc2
+    builtin and then its argument prototype would still apply.  */
+char $ac_func();
+
+int main() {
+
+/* The GNU C library defines this for functions which it implements
+    to always fail with ENOSYS.  Some functions are actually named
+    something starting with __ and the normal name is an alias.  */
+#if defined (__stub_$ac_func) || defined (__stub___$ac_func)
+choke me
+#else
+$ac_func();
+#endif
+
+; return 0; }
+EOF
+if { (eval echo configure:5001: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
+  rm -rf conftest*
+  eval "ac_cv_func_$ac_func=yes"
+else
+  echo "configure: failed program was:" >&5
+  cat conftest.$ac_ext >&5
+  rm -rf conftest*
+  eval "ac_cv_func_$ac_func=no"
+fi
+rm -f conftest*
+fi
+
+if eval "test \"`echo '$ac_cv_func_'$ac_func`\" = yes"; then
+  echo "$ac_t""yes" 1>&6
+    ac_tr_func=HAVE_`echo $ac_func | tr 'abcdefghijklmnopqrstuvwxyz' 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'`
+  cat >> confdefs.h <<EOF
+#define $ac_tr_func 1
+EOF
+ 
+else
+  echo "$ac_t""no" 1>&6
+fi
+done
+
+fi
+
 # Set up the CFLAGS for real compilation, so we can substitute it.
 CFLAGS="$REAL_CFLAGS"
 
@@ -5211,6 +5338,7 @@ s%@INSTALL_SCRIPT@%$INSTALL_SCRIPT%g
 s%@INSTALL_DATA@%$INSTALL_DATA%g
 s%@YACC@%$YACC%g
 s%@SET_MAKE@%$SET_MAKE%g
+s%@LIBXPM@%$LIBXPM%g
 s%@ALLOCA@%$ALLOCA%g
 s%@version@%$version%g
 s%@configuration@%$configuration%g
@@ -5445,6 +5573,7 @@ fi; done
 EOF
 cat >> $CONFIG_STATUS <<EOF
 CPP="$CPP" CPPFLAGS="$CPPFLAGS"
+MAKEFILECPP="$MAKEFILECPP"
 EOF
 cat >> $CONFIG_STATUS <<\EOF
 
@@ -5480,7 +5609,7 @@ echo creating lib-src/Makefile
   sed -e '1,/start of cpp stuff/d'\
       -e 's@/\*\*/#\(.*\)$@/* \1 */@' \
       < Makefile.c > junk.c
-  $CPP $undefs -I. -I$top_srcdir/src $CPPFLAGS junk.c | \
+  $MAKEFILECPP $undefs -I. -I$top_srcdir/src $CPPFLAGS junk.c | \
       sed -e 's/^ /	/' -e '/^#/d' -e '/^[ 	]*$/d' > junk2.c
   cat junk1.c junk2.c > Makefile.new
   rm -f junk.c junk1.c junk2.c
@@ -5496,7 +5625,7 @@ echo creating src/Makefile
   sed -e '1,/start of cpp stuff/d'\
       -e 's@/\*\*/#\(.*\)$@/* \1 */@' \
       < Makefile.c > junk.c
-  $CPP $undefs -I. -I$top_srcdir/src $CPPFLAGS junk.c | \
+  $MAKEFILECPP $undefs -I. -I$top_srcdir/src $CPPFLAGS junk.c | \
       sed -e 's/^ /	/' -e '/^#/d' -e '/^[ 	]*$/d' > junk2.c
   cat junk1.c junk2.c > Makefile.new
   rm -f junk.c junk1.c junk2.c
